src/tools/perf/lib/uct_tests.cc

```diff
@@ -398,16 +430,26 @@ class uct_perf_test_runner {
             } else{
                 send_sn     = 0; /* Remote buffer will remain 0 throughout the test */
             }
-            *(psn_t*)buffer = send_sn;
+
+            set_sn(&m_perf, (psn_t*)buffer, m_perf.uct.send_mem.mem_type,
+                   &send_sn, UCS_MEMORY_TYPE_HOST);
```

Comment:
the memory types (src and dest) can be kept as member variable. no need to pass them as args. also we can pas the buffer as void* / const void* to cast inside the getter/setter and avoid code duplication